airflow/providers/google/cloud/hooks/looker.py

```diff
@@ -0,0 +1,268 @@
+
+        :param materialization_id: Required. The materialization id to stop.
+        """
+        self.log.info("Stopping PDT materialization. Job id: %s.", materialization_id)
+        self.log.debug("PDT materialization job source: '%s'.", self.source)
+
+        sdk = self.get_looker_sdk()
+        resp = sdk.stop_pdt_build(materialization_id=materialization_id, source=self.source)
+
+        self.log.info("Stop PDT build response: '%s'.", resp)
+        return resp
+
+    def wait_for_job(
+        self,
+        materialization_id: str,
+        wait_time: int = 10,
+        timeout: Optional[int] = None,
+    ) -> None:
+        """
+        Helper method which polls a PDT materialization job to check if it finishes.
+
+        :param materialization_id: Required. The materialization id to wait for.
+        :param wait_time: Optional. Number of seconds between checks.
+        :param timeout: Optional. How many seconds wait for job to be ready.
+        Used only if ``asynchronous`` is False.
+        """
+        self.log.info('Waiting for PDT materialization job to complete. Job id: %s.', materialization_id)
+
+        status = None
+        start = time.monotonic()
+
+        while status not in (
+            JobStatus.DONE.value,
+            JobStatus.ERROR.value,
+            JobStatus.CANCELLED.value,
+            JobStatus.UNKNOWN.value,
+        ):
+
+            if timeout and start + timeout < time.monotonic():
+                raise AirflowException(
+                    f"Timeout: PDT materialization job is not ready after {timeout}s. "
+                    f"Job id: {materialization_id}."
+                )
+
+            time.sleep(wait_time)
+
+            status_dict = self.pdt_build_status(materialization_id=materialization_id)
+            status = status_dict['status']
+
+        if status == JobStatus.ERROR.value:
+            msg = status_dict['message']
+            raise AirflowException(
+                f'PDT materialization job failed. Job id: {materialization_id}. Message:\n"{msg}"'
+            )
+        if status == JobStatus.CANCELLED.value:
+            raise AirflowException(f'PDT materialization job was cancelled. Job id: {materialization_id}.')
+        if status == JobStatus.UNKNOWN.value:
+            raise AirflowException(
+                f'PDT materialization job has unknown status. Job id: {materialization_id}.'
+            )
+
+        self.log.info('PDT materialization job completed successfully. Job id: %s.', materialization_id)
+
+    def get_api_source(self):
+        """Returns origin of the API call."""
+
+        # Infer API source based on Airflow version, e.g.:
+        # "2.1.4+composer" -> Cloud Composer
+        # "2.1.4" -> Airflow
+
+        self.log.debug(
+            "Got the following version for connection id: '%s'. Version: '%s'.", self.looker_conn_id, version
+        )
+
+        return 'composer' if 'composer' in version else 'airflow'
```

Comment:
We don't want any SasS-specific code in our code. If you need to identify environments, you can use an environment variable and configure this parameter based on it.
For example, see: 
https://github.com/apache/airflow/blob/272d242316a1d163598d93511e2bfd7b717b4dce/airflow/providers/snowflake/hooks/snowflake.py#L184

Reply:
We'd like to track adoption of this integration. And we'd like to differentiate whether Looker API was called from a standalone Airflow or a managed one. Especially we are interested in Cloud Composer.

Reply:
I was hoping to achieve that in an effortless way for the users, that's why we decided to check the Airflow's `version` for additional info.

I'm not sure how an env var like `AIRFLOW_SNOWFLAKE_PARTNER` will be used. Does it mean that a user need to specify? If so, is there a ways to identify the type of Airflow (standalone/managed) based in a build-in data?

Reply:
>  Does it mean that a user need to specify it? 

Yes. The Cloud Composer team needs to add this variable in their Docker image. 
```
ENV AIRFLOW_GOOGLE_PARTNER=composer
```

> If so, is there a ways to identify the type of Airflow (standalone/managed) based in a build-in data?

No, because we are not involved in the development of a managed versions of Airflow and we want to be vendor-vendor neutral. This means that we cannot change the behavior of the code depending on which vendor is running the code. If a vendor wants to make changes to the behavior of the software, this change should only be made in the vendor's environment.

Reply:
Got it, I think I'll stick with tracking the `version` string itself then if that's okay? I don't want to introduce additional overhead right now for this small tracking feature.

Reply:
Tracking the version number is fine.